src/main.rs

```diff
@@ -508,6 +510,27 @@ fn run(args: Cli) -> Result<i32> {
 
         let mut request = request_builder.headers(headers).build()?;
 
+        if args.compress >= 1 && request.headers().get(CONTENT_ENCODING).is_none() {
+            let mut compressed = false;
+            if let Some(body) = request.body_mut() {
+                if let Ok(body_bytes) = body.buffer() {
```

Comment:
Looks like `.buffer()` only returns `Err` for IO errors, so `let body_bytes = body.buffer()?;` is probably better.